src/PixiEditor.ChangeableDocument/Changes/Root/CenterContent_Change.cs

```diff
@@ -0,0 +1,109 @@
+﻿using PixiEditor.ChangeableDocument.Changes.Drawing;
+using PixiEditor.DrawingApi.Core.Numerics;
+
+namespace PixiEditor.ChangeableDocument.Changes.Root;
+
+internal class CenterContent_Change : Change
+{
+    private VecI _oldOffset;
+    private List<Guid> _affectedLayers;
+    private Dictionary<Guid, CommittedChunkStorage>? originalLayerChunks;
+
+    [GenerateMakeChangeAction]
+    public CenterContent_Change(List<Guid> layers)
+    {
+        _affectedLayers = layers;
+    }
+    
+    public override bool InitializeAndValidate(Document target)
+    {
+        if (_affectedLayers.Count == 0)
+        {
+            return false;
```

Comment:
a tip: Returning false is very similar to throwing an exception, in a sense that it should never happen. So ideally we should only return false when something clearly doesn't make sense. I can see how pressing center content on a document with 0 layers would result in a validation failure here. So maybe we should handle it as a valid case (by doing nothing and setting ignoreInUndo to true)? tbh not sure, so this is just a tip

Edit: I see now that it the UI handles the "no layers selected" case so we can keep it as is